metaflow/plugins/cards/card_decorator.py

```diff
@@ -176,17 +199,20 @@ def task_pre_step(
         if self.step_counter == self.total_decos_on_step[step_name]:
             current.card._finalize()
 
-        self._task_datastore = task_datastore
-        self._metadata = metadata
-
     def task_finished(
         self, step_name, flow, graph, is_task_ok, retry_count, max_user_code_retries
     ):
-        if not is_task_ok:
-            return
-        component_strings = current.card._serialize_components(self._card_uuid)
-        runspec = "/".join([current.run_id, current.step_name, current.task_id])
-        self._run_cards_subprocess(runspec, component_strings)
+        create_options = dict(
+            card_uuid=self._card_uuid,
+            user_set_card_id=self._user_set_card_id,
+            runtime_card=self._is_runtime_card,
+            decorator_attributes=self.attributes,
+            card_options=self.card_options,
+            logger=self._logger,
+        )
+        if is_task_ok:
+            self.card_creator.create(mode="render", final=True, **create_options)
+            self.card_creator.create(mode="refresh", final=True, **create_options)
```

Comment:
WHy do we refresh after a final render?

Reply:
When we are viewing the card in the UI, the data update calls contain the information about what is the latest reload token; This helps the UI make a decision on weather it needs to re-fetch the cards or not. So during the final render, since we are consciously setting the reload token to `final`, we also want to ship a data update that can help information the UI that new cards need to be fetched again.